extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerTimedInterceptor.java

```diff
@@ -82,11 +71,40 @@ private Object processWithTimer(InvocationContext context, Timed timed, Tags com
             exceptionClass = MicrometerRecorder.getExceptionTag(ex);
             throw ex;
         } finally {
-            record(timed, metricName, sample, exceptionClass, timerTags);
+            for (Sample sample : samples) {
+                sample.stop(exceptionClass);
+            }
         }
     }
 
-    private void record(Timed timed, String metricName, Timer.Sample sample, String exceptionClass, Tags timerTags) {
+    private List<Sample> getSamples(InvocationContext context) {
```

Comment:
Thanks for cleaning this up. Started as a straight-up port of the Micrometer Aspect..